pkg/util/eventually.go

```diff
@@ -11,7 +11,7 @@ import (
 var backoffRetry = backoff.Config{
 	MinBackoff: 10 * time.Millisecond,
 	MaxBackoff: 1 * time.Second,
-	MaxRetries: 5,
+	MaxRetries: 10,
```

Comment:
I was intermittently getting the following in CI:
```
--- FAIL: Test_serviceManager (1.28s)
    --- FAIL: Test_serviceManager/can_run_service_binary (0.28s)
        eventually.go:48: 
            	Error Trace:	/Users/runner/work/agent/agent/cmd/grafana-agent-service/service_test.go:40
            	            				/Users/runner/work/agent/agent/cmd/grafana-agent-service/eventually.go:68
            	            				/Users/runner/work/agent/agent/cmd/grafana-agent-service/eventually.go:33
            	            				/Users/runner/work/agent/agent/cmd/grafana-agent-service/eventually.go:21
            	            				/Users/runner/work/agent/agent/cmd/grafana-agent-service/service_test.go:38
            	Error:      	Received unexpected error:
            	            	Post "http://127.0.0.1:49178/echo/response": dial tcp 127.0.0.1:49178: connect: connection refused
FAIL
FAIL	github.com/grafana/agent/cmd/grafana-agent-service	1.496s
```
Couldn't repro this locally. Potentially macOS CI agent is slow (they often are) and we need more time.